common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -640,14 +655,20 @@ def get_problem_html(self, encapsulate=True):
             check_button = False
             check_button_checking = False
 
-        problem_is_timed = self.minutes_allowed > 0
+        # For timed problems
+        now = datetime.datetime.now(UTC())
+        problem_has_finished = False
+        seconds_left = -1
+        end_time_to_display = now + datetime.timedelta(minutes=self.minutes_allowed)
```

Comment:
it seems like `end_time_to_display` isn't used for untimed problems, but I'd feel better if it was pretty much always right, which I don't think it is for untimed problems.  It seems to be set to `now` in those cases, and I'd prefer if it was `self.due`